### PR TITLE
[NN] Update GNNExplainer

### PR DESCRIPTION
**Pull request by mufeili**
## Description
Update GNNExplainer by exposing more arguments that users can tune.

It was inspired by [this thread](https://discuss.dgl.ai/t/gnnexplainer-import/2770/16).

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/nn/pytorch/explain/gnnexplainer.py [hunk 1]
```diff
@@ -35,28 +35,41 @@ class GNNExplainer(nn.Module):
         The learning rate to use, default to 0.01.
     num_epochs : int, optional
         The number of epochs to train.
+    edge_size : float, optional
```
Comment (jermainewang) [inline python/dgl/nn/pytorch/explain/gnnexplainer.py:38]: Is this some sort of budgets? It is quite counter-intuitive to have a higher "size" value that results in fewer edges.

File: python/dgl/nn/pytorch/explain/gnnexplainer.py [hunk 2]
```diff
@@ -35,28 +35,41 @@ class GNNExplainer(nn.Module):
...
-
     def __init__(self,
                  model,
                  num_hops,
                  lr=0.01,
                  num_epochs=100,
+                 edge_size=0.005,
+                 edge_ent=1.0,
+                 node_feat_size=1.0,
+                 node_feat_ent=0.1,
```
Comment (jermainewang) [inline python/dgl/nn/pytorch/explain/gnnexplainer.py:62]: I suggest making some of those arguments key-words only. E.g.

```python
    def __init__(self,
                 model,
                 num_hops,
                 lr=0.01,
                 num_epochs=100,
                 *,
                 edge_size=0.005,
                 edge_ent=1.0,
                 node_feat_size=1.0,
                 node_feat_ent=0.1,
```

File: python/dgl/nn/pytorch/explain/gnnexplainer.py (hunk 1 above)
Comment (mufeili) [reply python/dgl/nn/pytorch/explain/gnnexplainer.py:38]: It's the weight of a regularizer term for penalizing edge size. Any naming suggestions?

File: python/dgl/nn/pytorch/explain/gnnexplainer.py (hunk 2 above)
Comment (mufeili) [reply python/dgl/nn/pytorch/explain/gnnexplainer.py:62]: I'm fine with the proposal.

File: python/dgl/nn/pytorch/explain/gnnexplainer.py (hunk 1 above)
Comment (mufeili) [reply python/dgl/nn/pytorch/explain/gnnexplainer.py]: Updated

File: python/dgl/nn/pytorch/explain/gnnexplainer.py (hunk 2 above)
Comment (mufeili) [reply python/dgl/nn/pytorch/explain/gnnexplainer.py]: Updated

**Review by jermainewang — APPROVED** [review]

**Pull request merged**